beacon-chain/beacon-network.md

```diff
@@ -155,6 +171,9 @@ content                    = SSZList(ForkDigest + LightClientUpdate, max_lenght=
 content_key                = selector + SSZ.serialize(light_client_update_keys)
 ```
 
+> If a node cannot provide one of the `LightClientUpdate` objects in the
+the requested range it MUST NOT reply any content.
```

Comment:
Can we change this to just say it will send an empty TALKRESP?  That way we don't lose connections to otherwise "good actors" through discv5 banning a non-responsive peer.

Reply:
I consider an empty TALKRESP as a node that is not compatible. So I rather don't do that.

The node can perfectly be compatible with the protocol, but just not be synced yet.

Reply:
Right, so how should the node respond in this case?  With a FOUNDCONTENT message with an empty array?  I believe discv5 penalizes nodes who don't respond so if the responding node just doesn't respond, doesn't that lead to unnecessary peer bans?

Reply:
Hmm, If we don't respond to a talkreq, isn't this a violation of discv5 specs, where we must respond to any talkreq by talkresp?

Reply:
To comply to discv5 spec, we need to respond to the `talkreq` with a `talkresp`. Not responding here might/will make a node remove you from the discv5 routing table.
To indicate that we support the portal wire protocol for a specific network (beacon chain network in this case), we need to reply with the proper portal protocol response in the `talkreq` message. Responding empty here might/will make a node remove you from the specific portal network routing table as it is an indication that it does not support the specific talk protocol, see discv5 spec: `The recipient must respond with a TALKRESP message containing the response to the request. If the protocol is unknown to the recipient, it must respond with a TALKRESP response containing empty response data.`

What to do in this case is already explained in the PR here: https://github.com/ethereum/portal-network-specs/pull/225/files#diff-72c0d76cd767427aa91c94dfedd96c58a3b75d072203f8d2e6c70571e2438c6cR101-R103

This is also still in line with the portal wire protocol specification which states: `If the node does not hold the requested content, and the node does not know of any nodes with eligible ENR values, then the node MUST return enrs as an empty list.`

Reply:
Ah, sorry, I missed that.  I like that empty ENRs list response.  Not sure I'd ever noticed that in the spec before but it's a good solution.  Will implement that.